esphome/components/valve/valve.h

```diff
@@ -0,0 +1,152 @@
+#pragma once
+
+#include "esphome/core/component.h"
+#include "esphome/core/entity_base.h"
+#include "esphome/core/helpers.h"
+#include "esphome/core/preferences.h"
+#include "valve_traits.h"
+
+namespace esphome {
+namespace valve {
+
+const extern float VALVE_OPEN;
+const extern float VALVE_CLOSED;
+
+#define LOG_VALVE(prefix, type, obj) \
+  if ((obj) != nullptr) { \
+    ESP_LOGCONFIG(TAG, "%s%s '%s'", prefix, LOG_STR_LITERAL(type), (obj)->get_name().c_str()); \
+    auto traits_ = (obj) -> get_traits(); \
```

Comment:
Haha, why does it complain about this one line? 😅 
```suggestion
    auto traits_ = (obj)->get_traits(); \
```